silnlp/common/corpus.py

```diff
@@ -123,7 +123,10 @@ def filter_parallel_corpus(corpus: pd.DataFrame, score_threshold: float) -> pd.D
 
 
 def get_scripture_path(iso: str, project: str) -> Path:
-    return MT_SCRIPTURE_DIR / f"{iso}-{project}.txt"
+    scripture_file = MT_SCRIPTURE_DIR / f"{iso}-{project}.txt"
+    if not scripture_file.exists():
```

Comment:
Rather than checking if the file exists in these functions, I think it would be better to add these checks where the file is actually opened. These functions only build the path. There might be valid cases where the you want to build the path for a non-existent file. The code that you added in the `load_config` function is a good example of what I would suggest.

Reply:
Only checking when we try to open the file is causing me delays and is the main reason I'd like to put in the checks. In particular if the target data is missing I have to wait until the source vocabularies have been built before I find out.  It might be just a typo in the config but it happens often enough.  

An alternative could be a separate script or preprocess option that uses the current logic to find and report on the existence of all the files mentioned in a config file.

Reply:
You could still check if the file exists earlier. I just wouldn't check in these functions. It sounds like you want to check if all of the data files exist before preprocessing. I would suggest that perform the check at the beginning of the `Config.preprocess()` method in `config.py`. In that method, you could iterate through `src_file_paths` and `trg_file_paths` and check if each file exists. If a file doesn't exist, you would raise an exception.